.pre-commit-config.yaml

```diff
@@ -0,0 +1,27 @@
+exclude: |
+    (?x)^(
+        python/.*/auto_\w+/.*.py
+    )$
+fail_fast: false
+
+repos:
+  - repo: https://github.com/asottile/pyupgrade
+    rev: v3.19.0
+    hooks:
+      - id: pyupgrade
+        args: [--py39-plus]
+
+  - repo: https://github.com/psf/black
+    rev: 24.10.0
+    hooks:
+      - id: black
+
+  - repo: https://github.com/pycqa/isort
+    rev: 5.13.2
+    hooks:
+      - id: isort
+        args: ["--profile", "black", "--filter-files"]
+
+ci:
+  autofix_prs: true
+  autoupdate_schedule: quarterly
```

Comment:
Would that mean that if you're the first to push a PR after a hook update (let's say black rules for instance), you're gonna push a lot of modified files which are not related to your PR? Or is there an auto PR/commit pushed automatically on the repo on update?

Reply:
autoupdate is to update the tools used, so nothing related to the source code.

if there's a hook update, there's not that many changes brought. And in a PR, IIRC only the modified files are tested.